src/main/java/com/mongodb/jdbc/BsonTypeInfo.java

```diff
@@ -228,6 +291,8 @@ public static BsonTypeInfo getBsonTypeInfoByName(String typeName) throws SQLExce
                         return BSON_DOUBLE;
                     case 7:
                         return BSON_DECIMAL;
+                    case 8:
+                        return BSON_DOCUMENT;
```

Comment:
```suggestion
```
Since the mongodb [docs](https://www.mongodb.com/docs/manual/reference/bson-types/) and ADL use the name "object" instead of "document", I'm not sure this change is correct. Or at least, it may be dead code since we won't ever get a schema from ADL that uses the word "document" to describe this type.